### PR TITLE
feat(interactables): user picker component

### PR DESCRIPTION
**Pull request by jasonwoodland**
**What this PR does** 📖
- add new checkbox component
- add reusable friend picker component
- upgrade quick component to use new friend picker
- fix button animation speed when switching `type` prop in quick component
- add missing type definition

**Which issue(s) this PR fixes** 🔨
AP-1742
AP-1710

**Special notes for reviewers** 🗒️

**Additional comments** 🎤

### CODE REVIEW

File: components/interactables/UserPicker/UserPicker.vue [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  computed: {
+    ...mapState(['friends']),
+    filteredFriends() {
+      if (this.filter) {
+        const filterLower = this.filter.toLowerCase()
+        return this.friends.all.filter((friend: Friend) => {
+          return friend.name.toLowerCase().includes(filterLower)
+        })
+      }
+      return this.friends.all
```
Comment (josephmcg) [inline components/interactables/UserPicker/UserPicker.vue:23]: i think you can skip the guard clause, includes('') will be true

File: components/interactables/UserPicker/UserPicker.vue [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    },
+  },
+  watch: {
+    selected() {
+      console.log('emit', this.selected)
```
Comment (josephmcg) [inline components/interactables/UserPicker/UserPicker.vue:28]: remove logs

**Review by josephmcg — CHANGES_REQUESTED** [review]

File: components/interactables/UserPicker/UserPicker.vue (hunk 1 above)
Comment (jasonwoodland) [reply components/interactables/UserPicker/UserPicker.vue]: I feel if we can skip a superfluous filter call then we should from the perspective of performance.

**Review by josephmcg — APPROVED** [review]

**Comment by josephmcg** [comment]
making a note that we want to add some sort of list to keep track of currently selected users. thought it would be best in a separate PR to keep things simple

**Comment by jasonwoodland** [comment]
@phillsatellite fixed :)

**Comment by stavares843** [comment]
thanks @jasonwoodland 🔨

**Review by phillsatellite — APPROVED** [review]

**Comment by stavares843** [comment]
very neat 🎉 

i noticed that it doesn't show feedback if the search doesn't have results and the user tries to click on chat now even if the button is already not tinted but maybe we could improve that on other PR

Screenshot 2022-06-10 at 00 08 25

this branch vs dev where we don't show the button if there are no results

but this component is way prettier 🎉 

cc @josephmcg

**Review by stavares843 — APPROVED** [review]

File: components/interactables/UserPicker/UserPicker.html [hunk 3]
```diff
@@ -0,0 +1,16 @@
+<div class="user-picker">
```
Comment (stavares843) [inline components/interactables/UserPicker/UserPicker.html:1]: Captura de ecrã 2022-06-10, às 14 47 05

this is adding a warning when we build locally, could fix you on a separate PR? @jasonwoodland 🔨

Comment (stavares843) [reply components/interactables/UserPicker/UserPicker.html:1]: added ticket here - [AP-1766](https://satellite-im.atlassian.net/browse/AP-1766)

**Pull request merged**